gems/aws-crt/lib/aws-crt/auth/static_credentials_provider.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Aws
+  module Crt
+    module Auth
+      # CRT CredentialOptions
+      class StaticCredentialsProviderOptions
+        include Aws::Crt::ManagedNative
+        native_destroy Aws::Crt::Native.method(:credentials_provider_static_options_release)
+
+        # @param [String] access_key_id
+        # @param [String] secret_access_key
+        # @param [String] session_token (nil)
+        def initialize(access_key_id, secret_access_key,
+                       session_token = nil)
+          if !access_key_id || access_key_id.empty?
+            raise ArgumentError, 'access_key_id  must be set'
+          end
+
+          if !secret_access_key || secret_access_key.empty?
+            raise ArgumentError, 'secret_access_key  must be set'
+          end
```

Comment:
These were a function in another class. Does it make sense to have a shared interface like we do in V3 credentials?

Reply:
Maybe - I didn't want to directly connect these to any of our SDK interfaces.  These are providing just a basic validation that is required for the CRT bindings to work correctly.  IE: if you try to create a static credential provider without these set it will be in an invalid state - so we use some ruby code to enforce requiredness (hopefully a smithy model that describes these bindings would list required fields and we would essentially apply this logic blindly in the codegen'ed code for these bindings).

Reply:
That's fine - If sharing an interface is not ideal, I would prefer if there were in the validate method like you did in the other credentials object (to be consistent).